### PR TITLE
Feature/request partition

### PR DESCRIPTION
**Pull request by AndreaCuneo**

### CODE REVIEW

File: Artesian/Artesian.SDK/Service/Query/VersionedQuery.cs [hunk 1]
```diff
@@ -270,7 +274,15 @@ public VersionedQuery ForVersion(LocalDateTime version)
...
         public async Task<IEnumerable<TimeSerieRow.Versioned>> ExecuteAsync(CancellationToken ctk = default)
         {
-            return await _client.Exec<IEnumerable<TimeSerieRow.Versioned>>(HttpMethod.Get, _buildRequest(), ctk: ctk);
+            List<string> urls = _buildRequest();
+
+            var taskList = urls.Select(url => _client.Exec<IEnumerable<TimeSerieRow.Versioned>>(HttpMethod.Get, url, ctk: ctk));
+
+            await Task.WhenAll(taskList.ToArray());
+
+            var res = taskList.SelectMany(t => t.GetAwaiter().GetResult());
```
Comment (AndreaCuneo) [inline Artesian/Artesian.SDK/Service/Query/VersionedQuery.cs]: ELIMINATE any `GetResult()` from any async flow. Use await.
And btw WhenAll() returns what you need already, just use that.

If you ever use `GetResult()` again in an Async method, remember that for sure is wrong.

File: Artesian/Artesian.SDK/Service/Query/VersionedQuery.cs [hunk 2]
```diff
@@ -21,12 +22,15 @@ public class VersionedQuery : Query, IVersionedQuery<VersionedQuery>
         private VersionSelectionType? _versionSelectionType = null;
         private Granularity? _granularity;
         private Client _client;
+        private IPartitionStrategy _partition;
+
         private int? _tr;
         private string _routePrefix = "vts";
 
-        internal VersionedQuery(Client client)
+        internal VersionedQuery(Client client, IPartitionStrategy partiton)
```
[the remark this answers is not in the document]
Comment (MichaelGeraghty) [reply Artesian/Artesian.SDK/Service/Query/VersionedQuery.cs]: Do you want to enable this configurability as part of the IArtesianServiceConfig passed to the QueryService ctor or allow the user to implement their own partition strategy and pass it to the query service ctor?

File: Artesian/Artesian.SDK/Service/Clients/Client.cs [hunk 3]
```diff
@@ -85,6 +89,9 @@ public Client(IArtesianServiceConfig config, string Url)
             formatters.Add(_jsonFormatter);
             _formatters = formatters;
 
+            _resilienceStrategy = policy.ResillianceStrategy();
```
[the remark this answers is not in the document]
Comment (MichaelGeraghty) [reply Artesian/Artesian.SDK/Service/Clients/Client.cs]: Method renamed to GetResillianceStrategy()

File: Artesian/Artesian.SDK/Service/Configuration/ArtesianPolicyConfig.cs [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    {
+        private  AsyncCircuitBreakerPolicy _circuitBreakerPolicy { get; set; }
+
+        private  AsyncRetryPolicy _retryPolicy { get; set; }
+
+        private  AsyncBulkheadPolicy _bulkheadPolicy { get; set; }
+
+        private const int MaxParallelism = 10;
+        private const int MaxQueuingActions = 15;
+        private const int MaxExceptions = 2;
```
[the remark this answers is not in the document]
Comment (MichaelGeraghty) [reply Artesian/Artesian.SDK/Service/Configuration/ArtesianPolicyConfig.cs]: Removed MaxExceptions from Constants

File: Artesian/Artesian.SDK/Service/Configuration/ArtesianPolicyConfig.cs [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+        /// <summary>
+        /// Circuit Breaker Policy
+        /// </summary>
+        /// <param name="maxExceptions"></param>
+        /// <param name="durationOfBreak"></param>
+        /// <returns></returns>
+        public ArtesianPolicyConfig CircuitBreakerPolicyConfig(int maxExceptions = MaxExceptions, int durationOfBreak = DurationOfBreak)
+        {
+            _circuitBreakerPolicy = Policy
+                .Handle<AggregateException>(x =>
```
[the remark this answers is not in the document]
Comment (MichaelGeraghty) [reply Artesian/Artesian.SDK/Service/Configuration/ArtesianPolicyConfig.cs]: Handles Exception

File: Artesian/Artesian.SDK/Service/Configuration/ArtesianPolicyConfig.cs [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+        /// Bulkhead Policy Config
+        /// </summary>
+        /// <param name="maxParallelism"></param>
+        /// <param name="maxQueuingActions"></param>
+        /// <returns></returns>
+        public ArtesianPolicyConfig BulkheadPolicyConfig(int maxParallelism = MaxParallelism, int maxQueuingActions = MaxQueuingActions)
+        {
+            _bulkheadPolicy = Policy
+                .BulkheadAsync(
+                maxParallelism,
```
[the remark this answers is not in the document]
Comment (MichaelGeraghty) [reply Artesian/Artesian.SDK/Service/Configuration/ArtesianPolicyConfig.cs]: Wrong indentation resolved.

File: Artesian/Artesian.SDK/Service/Query/MasQuery.cs [hunk 7]
```diff
@@ -129,34 +132,51 @@ public MasQuery ForProducts(params string[] products)
...
         #region private
-        private string _buildRequest()
+        private List<string> _buildRequest()
         {
             _validateQuery();
 
             string url = null;
+            List<string> urlList = new List<string>();
+
+            var masParams = new MasQueryParamaters(_ids, _products);
```
[the remark this answers is not in the document]
Comment (MichaelGeraghty) [reply Artesian/Artesian.SDK/Service/Query/MasQuery.cs]: Extraction range and extraction range type passed.

File: Artesian/Artesian.SDK/Service/Query/VersionedQuery.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (MichaelGeraghty) [reply Artesian/Artesian.SDK/Service/Query/VersionedQuery.cs]: QueryService provides constructors that default to (CreateMarketAssessment()) PartitionByIDsStrategy() and (CreateMarketAssessment(IPartitionStrategy partitionStrategy))
_partition = partitionStrategy;

**Pull request closed without merging**